### ISSUE TITLE
"Cannot create data directory" on macOS Monterey

### ISSUE THREAD

**Issue by bram-dingelstad** [godotengine/godot#55368]
### Godot version

3.4.stable.official.206ba70f4

### System information

Apple M1 Pro, macOS Montery

### Issue description

When launching the project manager of Godot, it shows an empty screen even if the previous time you imported projects.
When launching Godot from CLI it shows the following:

```
0: /Applications/Godot 3.4.app/Contents/MacOS/Godot
Current path: /Users/bram
Godot Engine v3.4.stable.official.206ba70f4 - https://godotengine.org
OpenGL ES 3.0 Renderer: Apple M1 Pro
OpenGL ES Batching: ON

UNSUPPORTED (log once): POSSIBLE ISSUE: unit 1 GLD_TEXTURE_INDEX_2D is unloadable and bound to sampler type (Float) - using zero texture because texture unloadable
Registered camera FaceTime HD Camera with id 1 position 0 at index 0
ERROR: Cannot create data directory!
   at: create (editor/editor_settings.cpp:871)
ERROR: Cannot save EditorSettings config, no valid path
   at: save (editor/editor_settings.cpp:1101)
ERROR: Cannot save EditorSettings config, no valid path
   at: save (editor/editor_settings.cpp:1101)
   ```
   
This is on a brand new MacBook Pro M1 Pro (10 core). I'm gonna go on hunt to see if there anything I can do to fix this temporarily, but wanted to report it in case someone else ran into it.

### Steps to reproduce

1. Download Godot v3.4 for Apple Silicon. 
2. Execute Godot v3.4 with CLI (e.g `/Applications/Godot.app/Contents/MacOS/Godot`)
3. Look at CLI `stderr`

### Minimal reproduction project

Not needed to reproduce.

**Comment by bram-dingelstad** [comment]
I tried the same reproduction steps but with v3.4.1 and v3.3.4, still the same error:

```
Current path: /Users/bram/Downloads
Godot Engine v3.4.1.rc1.official.7b0801c7f - https://godotengine.org
OpenGL ES 3.0 Renderer: Apple M1 Pro
OpenGL ES Batching: ON

UNSUPPORTED (log once): POSSIBLE ISSUE: unit 1 GLD_TEXTURE_INDEX_2D is unloadable and bound to sampler type (Float) - using zero texture because texture unloadable
Registered camera FaceTime HD Camera with id 1 position 0 at index 0
ERROR: Cannot create data directory!
   at: create (editor/editor_settings.cpp:871)
ERROR: Cannot save EditorSettings config, no valid path
   at: save (editor/editor_settings.cpp:1101)
 ```
 
 and
 
 ```
 Godot Engine v3.3.4.stable.official.faf3f883d - https://godotengine.org
OpenGL ES 3.0 Renderer: Apple M1 Pro
OpenGL ES Batching: ON

UNSUPPORTED (log once): POSSIBLE ISSUE: unit 1 GLD_TEXTURE_INDEX_2D is unloadable and bound to sampler type (Float) - using zero texture because texture unloadable
Registered camera FaceTime HD Camera with id 1 position 0 at index 0
ERROR: create: Cannot create data directory!
   At: editor/editor_settings.cpp:900.
ERROR: save: Cannot save EditorSettings config, no valid path
   At: editor/editor_settings.cpp:1107.
   ```

**Comment by bram-dingelstad** [comment]
~~I've found out that apparently on the first run it creates a file with the same name as the `data_dir` for macOS (aka `~/Library/Application Support/Godot`). In here is what looks like the state file for recording editor state.~~

OKAY!!! It turns out that I actually created this myself by manually creating a symlink to sync my Godot settings. It actually created a file at the `data_dir` location. Case closed! My apologies for starting an issue for it. But perhaps this issue can help other people that might unknowningly run into the same issue.

Moral of the story: make sure that you always use `mkdir -p` on locations you expect to exist in your scripts 😅
